packages/lit-html/src/lit-html.ts

```diff
@@ -400,18 +392,13 @@ class Template {
         }
       }
 
-      if (DEV_MODE) {
-        // We should never be in both text position (regex === textRegex) and
-        // attribute position (attrNameEnd !== -1).
-        console.assert(!(attrNameEnd !== -1 && regex === textRegex));
-      }
-      if (attrNameEnd !== -1) {
-        attrNames.push(attrName!);
-        html +=
-          s.slice(0, attrNameEnd) + '$lit$' + s.slice(attrNameEnd) + marker;
-      } else {
-        html += regex === textRegex ? s + nodeMarker : s + marker;
-      }
+      html +=
+        regex === textRegex
+          ? s + nodeMarker
+          : (attrNameEnd !== -1
```

Comment:
I think we'd need to reset the `attrNameEnd`, or inline it in the for-loop? Else a `foo="${part1} ${part2} ${part3}"` might get weird. I think `attrNames` will contain `['foo', ' ', ' ']`, and the `<template>` will have `$lit$`s where it shouldn't.

Reply:
`attrNameEnd` is a loop-local variable, we had mistaken code in here assuming it wasn't 🤫

Reply:
Ohh, for some reason it appeared above the for-loop for me. This should be fine, then.

Reply:
Yeah, we had the same confusion for a second

Reply:
Let's add a comment explaining this change. We discussed refining this more and the comment will probably help us do that.